### ISSUE TITLE
[a11y]: Popover [autoalign] - update heading semantic

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#14823]
### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.38.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO & JAWS

### Description
The text "Available storage" That visually looks prominent like heading

Heading role not Assigned

### Recommended Fix
Screen reader announces the mentioned text only as A plain text. The heading Role is not announced

Provide appropriate heading tag

When structural markups like headings and lists are not used, Screen readers cannot convey the semantic information of those elements. This impacts users who are blind

### WCAG 2.1 Violation

1.3.1 Info and Relationships
